lib/inspec/plugins/resource.rb

```diff
@@ -37,14 +37,24 @@ def __resource_registry
       Inspec::Resource.registry
     end
 
-    def __register(name, obj)
+    def __register(name, obj) # rubocop:disable Metrics/MethodLength
       cl = Class.new(obj) do
         def initialize(backend, name, *args)
+          @resource_skipped = nil
```

Comment:
Should we use an enum? Essentially we have stopped the execution and we did this for different reasons. The reason is important to say if this is okay or not.